Added functions dashboard quickstart page, about deno 2 guide, and deno 2 test guide (#34599)
supabase/supabase: apps/docs/content/guides/functions/quickstart.mdx

---
id: 'functions-quickstart'
title: 'Developing Edge Functions with Supabase'
description: 'Get started with Edge Functions on the Supabase dashboard.'
subtitle: 'Get started with Edge Functions on the Supabase dashboard.'
tocVideo: '5OWH9c4u68M'
---

In this guide we'll cover how to create a basic Edge Function on the Supabase dashboard, and access it using the Supabase CLI.

## Deploy from Dashboard

Go to your project > Edge Functions > Deploy a new function > Via Editor

<Image
  alt="Deploy functions from the dashboard"
  src={{
    light: '/docs/img/guides/functions/dashboard/deploy-function--light.png',
    dark: '/docs/img/guides/functions/dashboard/deploy-function--light.png',
  }}
  zoomable
/>

> This will scaffold a new function for you.

Reviewer: @sayil Can we update to say they can select different templates?

<Image
  alt="Scaffold functions through the dashboard editor"
  src={{
    light: '/docs/img/guides/functions/dashboard/scaffold-function--light.png',
    dark: '/docs/img/guides/functions/dashboard/scaffold-function--light.png',
  }}
  zoomable
/>

Modify the function as needed, name it, and click `Deploy function`

> That's it! Your function is now active. Use the endpoint on the function's Details page to call the edge function.

Reviewer: We should suggest using Test option in the Dashboard to test the deployed function. Maybe a screenshot of Test tool is good.

      dark: '/docs/img/guides/functions/dashboard/test-request-body--light.png',
    }}
    zoomable
  />
</div>

## Editing functions from the Dashboard

<Admonition type="caution" label="Be careful: there is currently no version control for edits">

The Dashboard's Edge Function editor currently does not support versioning or rollbacks. We recommend using it only for quick testing and prototypes. When you’re ready to go to production, store Edge Functions code in a source code repository (e.g., git) and deploy it using one of the [CI integrations](https://supabase.com/docs/guides/functions/cicd-workflow).

</Admonition>

1. From the functions page, click on the function you want to edit. From the function page, click on the Code tab.

2. This opens up a code editor in the dashboard where you can see your deployed function's code.

3. Modify the code as needed, then click Deploy updates. This will overwrite the existing deployment with the newly edited function code.

## Next steps

Check out the [Local development](/docs/guides/functions/local-quickstart) guide for more details on working with Edge Functions.

> Read on for some common development tips.

Reviewer: Shall we move the development tips to a separate page? That makes Quickstart look succinct. Also, currently they are repeated on both Quickstart and Local Development. Better to keep them in a single and link to from other pages.

---
id: 'functions-quickstart'
title: 'Developing Edge Functions with Supabase'
description: 'Get started with Edge Functions on the Supabase dashboard.'
subtitle: 'Get started with Edge Functions on the Supabase dashboard.'
---

In this guide we'll cover how to create a basic Edge Function on the Supabase dashboard, and access it using the Supabase CLI.

## Deploy from Dashboard

Go to your project > Edge Functions > Deploy a new function > Via Editor

<Image
  alt="Deploy functions from the dashboard"
  src={{
    light: '/docs/img/guides/functions/dashboard/deploy-function--light.png',
    dark: '/docs/img/guides/functions/dashboard/deploy-function--light.png',
  }}
  zoomable
/>

This will scaffold a new function for you. You can choose from Templates some of the pre-defined functions for common use cases.

<Image
  alt="Scaffold functions through the dashboard editor"
  src={{
    light: '/docs/img/guides/functions/dashboard/scaffold-function--light.png',
    dark: '/docs/img/guides/functions/dashboard/scaffold-function--light.png',
  }}
  zoomable
/>

Modify the function as needed, name it, and click `Deploy function`

> That's it! Your function is now active. Navigate to the function's details page, and click on the test button.

Reviewer 2: ```suggestion
Your function is now active. Navigate to the function's details page, and click on the test button.
```

> You can test your function by providing expected HTTP method, headers, query parameters, and request body. You can also easily change the authorization token passed (eg: anon key or a user key).

Reviewer 2: ```suggestion
You can test your function by providing the expected HTTP method, headers, query parameters, and request body. You can also easily change the authorization token passed (e.g., anon key or a user key).
```

      dark: '/docs/img/guides/functions/dashboard/test-request-body--light.png',
    }}
    zoomable
  />
</div>

## Access deployed functions via Supabase CLI

<Admonition type="tip" label="CLI not installed?">

Check out the [CLI Docs](/docs/guides/cli) to learn how to install the Supabase CLI on your local machine.

</Admonition>

Now that your function is deployed, you can access it from your local development environment.
Here's how:

1. **Link your project** to your local environment.

   You can find your project reference ID in the URL of your Supabase dashboard or in the project settings.

   ```bash
   supabase link --project-ref your-project-ref
   ```

2. **List all Functions** in the linked Supabase project.

   ```bash
   supabase functions list
   ```

3. **Access the specific function** you want to work on.

   ```bash
   supabase functions download function-name
   ```

4. **Make local edits** to the function code as needed.

5. **Run your function locally** before redeploying.

   ```bash
   supabase functions serve function-name
   ```

6. **Redeploy** when you're ready with your changes.

   ```bash
   supabase functions deploy function-name
   ```

> ## Deploy via Assistant

Reviewer 2: ```suggestion
{/* supa-mdx-lint-disable-next-line Rule001HeadingCase */}

## Deploy via Assistant
```
